app/docs/async-api.yaml

```diff
@@ -0,0 +1,80 @@
+asyncapi: 2.4.0
+info:
+  title: FFC Payment Statement Constructor
+  version: '1.0.0'
+  description: Construct data for Payment Statements
+channels:
+  reurn/message:
+    subscribe:
+      message:
+        $ref: 
+components:
+  messages:
+    ReturnMessage:
+      name: Return Message
+      contentType: application/json
+      payload:
+        $ref: 
+  schemas:
+    ReturnMessage:
+      type: object
+      required:
+      - sourceSystem
+      - invoiceNumber
+      - frn
+      - currency
+      - value
+      - settlementDate
+      - reference 
+      - settled
+      properties:
+        sourceSystem:
+          $ref:/
+        invoiceNumber:
+          $ref:/
+        frn:
+          $ref: /
+        currency:
+          $ref:/
+        value:
+          $ref:/
+        settlementDate:
+          $ref:/
+        reference:
+            $ref:/
+        settled:
+            $ref:/
+    SourceSystem:
```

Comment:
so this `SourceSystem` thing is under `schema`

this is where we define the schema for `SourceSystem`, i.e. characteristics of it, e.g. its datatype, an example

we can then 'reference' this schema elsewhere in this doc, i.e. say that this message has a key called 'sourceSystem' and the value for that is defined by the `SourceSystem` schema given in this file, i.e. the value is of type string, example of SitiAgri

this gives us the ability to define the schema once and re-use many things in this doc


so for this document we need to define schemas for all the keys that a return message contains:

- currency
- detail
- frn
- invoiceNumber
- reference
- settled
- settlementDate
- sourceSystem
- value

most (if not all) have been defined in other asyncapi docs, re-use them :)

you might see in other asyncapi docs `valueDecimal` and `valueInteger` -- why do you think this is the case? how are they referenced in their respected asyncapi docs? how does effect your implementation of building your asyncapi doc, which do you use: either, both, neither?

we also need a schema for the `ReturnMessage` itself which we have on line 19 and this is where we reference those component schemas for current, detail, frn, etc, etc


havva go defining your `ReturnMessage` schema if the message coming in only had a `currency` field that is required

i.e. message coming onto `ffc-pay-return` looked like this

{
   "currency":"GBP"
}